web/src/unread.ts

```diff
@@ -215,24 +244,84 @@ class UnreadDirectMessageCounter {
 }
 const unread_direct_message_counter = new UnreadDirectMessageCounter();
 
-function make_per_stream_bucketer() {
+function make_per_stream_bucketer(): Bucketer {
     return new Bucketer({
         key_to_bucket: new FoldDict(), // bucket keys are topics
         make_bucket: () => new Set(),
     });
 }
 
+class UnreadTopicBucketer {
```

Comment:
This shouldn’t be separate; it’s just `Bucketer<number, Bucketer<string, number>>`.

Reply:
Oh I see, there are some problems with how this was designed.

Reply:
Tim suggested I remove this, but it makes the code a lot more gross. So I think it might make more sense to keep it for now and maybe redesign it later? I guess I can also try redesigning it but I'm less confident in that. I sent you a message to maybe pair on this (and some other remaining details)